test/common/integration/xds_integration_test.cc

```diff
@@ -0,0 +1,187 @@
+  EXPECT_TRUE(response->complete());
+  EXPECT_EQ("200", response->headers().getStatusValue());
+  Json::ObjectSharedPtr loader = TestEnvironment::jsonLoadFromString(response->body());
+  auto entries = loader->getObject("entries");
+  if (entries->hasObject(key)) {
+    return entries->getObject(key)->getString("final_value");
+  }
+  return "";
+}
+
+// TODO(abeyad): Change this implementation to use the PulseClient once implemented. See
+// https://github.com/envoyproxy/envoy-mobile/issues/2356 for details.
+uint64_t XdsIntegrationTest::getCounterValue(const std::string& counter) {
+  auto response = IntegrationUtil::makeSingleRequest(lookupPort("admin"), "GET", "/stats?usedonly",
+                                                     "", Http::CodecType::HTTP2, version_);
+  EXPECT_TRUE(response->complete());
+  EXPECT_EQ("200", response->headers().getStatusValue());
+  std::stringstream ss(response->body());
+  std::string line;
+  while (std::getline(ss, line, '\n')) {
+    auto pos = line.find(':');
+    if (pos == std::string::npos) {
+      continue;
+    }
+    if (line.substr(0, pos) == counter) {
+      return std::stoi(line.substr(pos + 1));
+    }
+  }
+  return 0;
+}
+
+AssertionResult XdsIntegrationTest::waitForCounterGe(const std::string& name, uint64_t value) {
+  constexpr std::chrono::milliseconds timeout = TestUtility::DefaultTimeout;
+  Event::TestTimeSystem::RealTimeBound bound(timeout);
+  while (getCounterValue(name) < value) {
+    Event::GlobalTimeSystem().timeSystem().advanceTimeWait(std::chrono::milliseconds(10));
+    if (timeout != std::chrono::milliseconds::zero() && !bound.withinBound()) {
+      return AssertionFailure() << fmt::format("timed out waiting for {} to be {}", name, value);
+    }
+  }
+  return AssertionSuccess();
+}
+
+envoy::config::cluster::v3::Cluster
+XdsIntegrationTest::createSingleEndpointClusterConfig(const std::string& cluster_name) {
+  envoy::config::cluster::v3::Cluster config;
+  config.set_name(cluster_name);
+
+  // Set the endpoint.
+  auto* load_assignment = config.mutable_load_assignment();
+  load_assignment->set_cluster_name(cluster_name);
+  auto* endpoint = load_assignment->add_endpoints()->add_lb_endpoints()->mutable_endpoint();
+  endpoint->mutable_address()->mutable_socket_address()->set_address(loopbackAddr());
+  endpoint->mutable_address()->mutable_socket_address()->set_port_value(0);
+
+  // Set the protocol options.
+  envoy::extensions::upstreams::http::v3::HttpProtocolOptions options;
+  options.mutable_explicit_http_config()->mutable_http2_protocol_options();
+  (*config.mutable_typed_extension_protocol_options())
+      ["envoy.extensions.upstreams.http.v3.HttpProtocolOptions"]
+          .PackFrom(options);
+  return config;
+}
+
+envoy::config::bootstrap::v3::Admin XdsIntegrationTest::adminConfig() {
+  const std::string yaml = fmt::format(R"EOF(
+    access_log:
+    - name: envoy.access_loggers.file
+      typed_config:
```

Comment:
does config require having an access log or can we just remove this section of config?

Reply:
Removed